test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -85,6 +83,65 @@ public void ItPublishesARunnableSelfContainedApp()
                      .And.HaveStdOutContaining("Hello World");
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ItPublishesAnAppExplicitlySpecifyingSelfContained(bool selfContained)
+        {
+            var testAppName = "MSBuildTestApp";
+
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance($"PublishesSelfContained{selfContained}")
+                .WithSourceFiles()
+                .WithRestoreFiles();
+
+            var testProjectDirectory = testInstance.Root;
+
+            var rid = DotnetLegacyRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier();
+
+            new PublishCommand()
+                .WithRuntime(rid)
+                .WithSelfContained(selfContained)
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should().Pass();
+
+            var configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+            var outputPath = testProjectDirectory
+                    .GetDirectory("bin", configuration, "netcoreapp2.0", rid, "publish")
+                    .FullName;
+            var selfContainedProgram = Path.Combine(outputPath, $"{testAppName}{Constants.ExeSuffix}");
+            var selfContainedProgramFile = new FileInfo(selfContainedProgram);
+
+            TestCommand testCommand;
+            string testArgs;
+            if (selfContained)
```

Comment:
I find that if there's an if statement in a test, it should probably be two different tests with more specific names. This will make the tests more informative as to the expected behavior, and the code in each test will be shorter and easier to understand.